From 0bef5378d18f3ce2d8028151d8fb683beee4ad00 Mon Sep 17 00:00:00 2001
From: Aisheng.Dong <b29396@freescale.com>
Date: Tue, 14 Sep 2010 14:23:01 +0800
Subject: [PATCH] ENGR00131466-8 mxs spi: calculate the actual_length transferred.

If we do not do actual_length calculation, the upper layer of spi
may fail when performing data sanity check such as m25p80 spi nor.

Signed-off-by: Aisheng.Dong <b29396@freescale.com>
---
 drivers/spi/spi_mxs.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/spi/spi_mxs.c b/drivers/spi/spi_mxs.c
index 149d24a..a6ca061 100644
--- a/drivers/spi/spi_mxs.c
+++ b/drivers/spi/spi_mxs.c
@@ -426,6 +426,7 @@ static int mxs_spi_handle_message(struct mxs_spi *ss, struct spi_message *m)
 			}
 		}
 
+		m->actual_length += t->len;
 		if (status)
 			break;
 
@@ -480,6 +481,7 @@ static int mxs_spi_transfer(struct spi_device *spi, struct spi_message *m)
 	struct mxs_spi *ss = spi_master_get_devdata(spi->master);
 	unsigned long flags;
 
+	m->actual_length = 0;
 	m->status = -EINPROGRESS;
 	spin_lock_irqsave(&ss->lock, flags);
 	list_add_tail(&m->queue, &ss->queue);
-- 
1.5.4.4

